Src/GoogleApis.Auth/OAuth2/ComputeCredential.cs

```diff
@@ -99,5 +120,53 @@ public Initializer(string tokenUrl)
         }
 
         #endregion
+
+        /// <summary>
+        /// Detects if application is running on Google Compute Engine. This is achieved by attempting to contact
+        /// GCE metadata server, that is only available on GCE. The check is only performed the first time you
+        /// call this method, subsequent invocations used cached result of the first call.
+        /// </summary>
+        public static Task<bool> IsRunningOnComputeEngine()
+        {
+            return isRunningOnComputeEngineCached.Value;
+        }
+
+        private static async Task<bool> IsRunningOnComputeEngineNoCache()
+        {
+            try
+            {
+                Logger.Info("Checking connectivity to ComputeEngine metadata server.");

```

Comment:
can we please limit the catch area to the ones that can actually throw exception?